js/main.js

```diff
@@ -58,8 +72,46 @@
     mapPins.appendChild(pinsDocumentFragment);
   }
 
-  mapBlock.classList.remove('map--faded'); // Визуально переводим карту в активное состояние
-  appendPinsFromDataArray(getMockAdsData()); // Выводим пины на карту
+  // Разблокирует формы
+  function enableForms() {
+    var enableElement = function (element) {
+      element.removeAttribute('disabled');
+    };
+    mapBlock.classList.remove('map--faded');
+    addForm.classList.remove('ad-form--disabled');
+    mapFilterFields.forEach(enableElement);
+    mapFeaturesFieldset.removeAttribute('disabled');
+    addFormHeader.removeAttribute('disabled');
+    addFormElements.forEach(enableElement);
+    isPageActive = true;
+  }
+
+  // Возвращает строку с текущими координатами пина в формате "left, top"
+  function getCurrentMainPinPosition() {
+    var coordX = Number(mapPinMain.style.left.slice(0, -2)) + MAIN_PIN_WIDTH / 2;
+    var coordY = Number(mapPinMain.style.top.slice(0, -2)) + MAIN_PIN_HEIGHT;
```

Comment:
```suggestion
    var coordY = mapPinMain.offsetTop + MAIN_PIN_HEIGHT;
```

можно попроще :) То же самое по горизонтали